packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -33,6 +44,62 @@ export class ForceSourceDeploySourcePathExecutor extends BaseDeployExecutor {
     return commandBuilder.build();
   }
 
+  public async execute(response: ContinueResponse<string>): Promise<void> {
+    const betaPerfEnabled = sfdxCoreSettings.getBetaPerfEnhancements();
+    // this supported types logic is temporary until we have a way of generating the metadata type from the path
+    // once we have the metadata type we can check to see if it is a toolingsupportedtype from that util
+    const supportedType =
+      path.extname(response.data) === APEX_CLASS_EXTENSION ||
+      response.data.includes(`${APEX_CLASS_EXTENSION}-meta.xml`);
+    if (betaPerfEnabled && supportedType) {
+      const startTime = process.hrtime();
+      const cancellationTokenSource = new vscode.CancellationTokenSource();
+      const cancellationToken = cancellationTokenSource.token;
+
+      const executionWrapper = new CompositeCliCommandExecutor(
+        this.build(response.data)
+      ).execute(cancellationToken);
+      this.attachExecution(
+        executionWrapper,
+        cancellationTokenSource,
+        cancellationToken
+      );
+      executionWrapper.processExitSubject.subscribe(() => {
+        this.logMetric('force_source_deploy_with_sourcepath_beta', startTime);
+      });
+
+      try {
+        const usernameOrAlias = await OrgAuthInfo.getDefaultUsernameOrAlias(
+          true
+        );
+        if (!usernameOrAlias) {
+          throw new Error(nls.localize('error_no_default_username'));
+        }
+        const orgConnection = await OrgAuthInfo.getConnection(usernameOrAlias);
+        const deployLibrary = new ToolingDeploy(orgConnection);
+        const deployOutput = await deployLibrary.deploy(response.data);
+
+        const parser = new ToolingDeployParser(deployOutput!);
+        await parser.outputResult(executionWrapper);
+        await DeployQueue.get().unlock();
```

Comment:
Can unlock the deploy queue in a `finally {}` block instead of having to do it in both the `try` and `catch`